### PR TITLE
Add last success column to the TG home page

### LINKED ISSUE #797: Remove the latest build column from the home page and add the last success column
**Description:**
As per an improvement, remove the last build column which displays timestamp of the last build from the TG home page and includes a new column, in order to display the timestamp of the last success. In TG we are displaying status and the timestamp of the last build in the deployment-pattern page. 

**Affected Product Version:**
0.9.30

### PR DESCRIPTION
**Pull request by lasanthaDLPDS**
**Purpose**
The purpose of this PR is to add last success field into TG home page. This resolves #797.

**Goals**
In order to increase the sense of displaying data on the home page, removed the latest build column from the home page and add the last success column. To maintain the product status modified the finalize-test-plan command. Loop all test plan files and get status of the test plan. If one of the test plans is in the FAIL status update the last_failure_timestamp column in the product table. If all test plans are in success status update the last_success_timestamp.

**Approach**
tg1

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

**Test environment**
>JDK - Oracle 1.8
>OS - UBUNTU 18.04
>DB - MySQL 5.7
>Browser - Chrome

### CODE REVIEW

File: dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java [hunk 1]
```diff
@@ -147,4 +149,37 @@ public Product findByPrimaryKey(String id) throws TestGridDAOException {
...
+     * The column is selected by considering product status.
+     *
+     * @param status Status of the product
+     * @param timestamp Current timestamp
+     * @param productId Id of the product
+     */
+    public void updateProductStatusTimestamp(Status status, Date timestamp, String productId)
+            throws TestGridDAOException {
+        try {
+            String queryStr = "UPDATE product SET ";
```
Comment (harshanL) [inline dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java]: Please use the StringUtil class to build the query string.

File: dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java [hunk 2]
```diff
@@ -147,4 +149,37 @@ public Product findByPrimaryKey(String id) throws TestGridDAOException {
...
+            queryStr += "'" + timestamp + "' where id = '" + productId + "';";
+
+            // Begin entity manager transaction
+            entityManager.getTransaction().begin();
+
+            entityManager.createNativeQuery(queryStr).executeUpdate();
+
+            // Commit transaction
+            entityManager.getTransaction().commit();
+        } catch (Exception e) {
```
Comment (sameerawickramasekara) [inline dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java]: Shall we catch a specific Exception here :)

File: core/src/main/java/org/wso2/testgrid/core/command/FinalizeRunTestplan.java [hunk 3]
```diff
@@ -143,4 +161,48 @@ private void persistTestPlan(TestPlan testPlan) {
...
         }
     }
+
+    /**
+     * Update the last success timestamp of the product build or last failure timestamp of the product build
+     * by considering status of test plans.
+     *
+     */
+    private void updateProductStatus() throws TestGridException {
+        Path source = Paths.get(workspace + "/test-plans");
```
Comment (kasunbg) [inline core/src/main/java/org/wso2/testgrid/core/command/FinalizeRunTestplan.java]: *Paths.get(workspace, "test-plans")

File: dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java [hunk 4]
```diff
@@ -147,4 +149,37 @@ public Product findByPrimaryKey(String id) throws TestGridDAOException {
         }
         return productTestStatuses;
     }
+
+    /**
+     * This method updates the last_success_timestamp column or last_failure_timestamp column with given timestamp.
+     * The column is selected by considering product status.
```
Comment (kasunbg) [inline dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java]: Please provide a sample sql query you are using for reference purposes.. It's a bit hard to follow the string construction logic.

Also, please run _FindSecurityBugs_ for this, and make sure no issues pops up. You can do this with IntelliJ Idea itself.

Comment (lasanthaDLPDS) [reply dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java]: Fixed with https://github.com/wso2/testgrid/pull/803/commits/992f5f36ed390df948dff65148704178a4dbe307
